tools/osv-linter/internal/checks/checks.go

```diff
@@ -0,0 +1,140 @@
+// Package checks defines and implements all checks and collections of checks.
+//
+// To add additional checks:
+// 1. define a new instance of `Check`
+// 2. add it to the `checks` array
+// 3. add it to the relevent collections defined in `checkCollections`
+//
+// To add additional collections of checks:
+// 1. add to the `checkCollections` array.
+package checks
+
+import (
+	"fmt"
+
+	"github.com/tidwall/gjson"
+)
+
+// A CheckCode is a unique code for a check.
+type CheckCode string
+
+// CheckError describes when a check fails.
+type CheckError struct {
+	Code    CheckCode
+	Message string
+}
+
+// Error returns the error message, including the code.
+func (ce *CheckError) Error() string {
+	return fmt.Sprintf("%s: %s", ce.Code, ce.Message)
+}
+
+// CodeString returns just the error code, as a string.
+func (ce *CheckError) CodeString() string {
+	return string(ce.Code)
+}
+
+// Checkers are for running a discrete checking function.
+type Checker interface {
+	CodeString() string
```

Comment:
Can you add some comments describing what these methods do? 

In particular, it's not clear how CodeString() and Name() differ just from this.